.github/workflows/build-and-test.yml

```diff
@@ -26,8 +27,17 @@ env:
   # SCCACHE_RECACHE: 1 # uncomment to clear sccache cache, then re-comment
 
 jobs:
+  check-for-run-condition:
+    runs-on: ubuntu-latest
+    outputs:
+      should-run: ${{ !github.event.pull_request || github.event.pull_request.draft == false }}
+    steps:
+      - run: |
+          # this run step does nothing, but is needed to get the job output
 
   check-for-modification:
+    needs: check-for-run-condition
+    if: ${{ needs.check-for-run-condition.outputs.should-run == 'true' }}
```

Comment:
I was thinking more along the lines of placing the condition in the `if` section of `check-for-run-condition`. Then add it to `needs` for the rest of the jobs. That way if it's skipped the rest of the jobs are skipped too (if I understand correctly).

E.g.
```yml
  check-for-run-condition:
    runs-on: ubuntu-latest
    if: ${{ !github.event.pull_request || github.event.pull_request.draft == false }}
    steps:
      - run: |
          # this run step does nothing, but is needed to get the job output

  check-for-modification:
    needs: check-for-run-condition
```

Reply:
Oh, for me that feels unidiomatic. In my understanding `needs` orders the jobs. It is meant to handle success and failure (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds), I think absence / skipped is undefined. Another possibility I looked into was putting the result of the condition in a env variable, but they can't be evaluated in the context of `if` .